pkg/services/grafana-apiserver/auth/authorizer/org/org_id.go

```diff
@@ -29,22 +34,46 @@ func (auth OrgIDAuthorizer) Authorize(ctx context.Context, a authorizer.Attribut
 		return authorizer.DecisionDeny, fmt.Sprintf("error getting signed in user: %v", err), nil
 	}
 
-	orgID, ok := grafanarequest.ParseOrgID(a.GetNamespace())
-	if !ok {
+	info, err := grafanarequest.ParseNamespace(a.GetNamespace())
+	if err != nil {
+		return authorizer.DecisionDeny, fmt.Sprintf("error reading namespace: %v", err), nil
+	}
+
+	// No opinion when the namespace is arbitrary
+	if info.OrgID == -1 {
 		return authorizer.DecisionNoOpinion, "", nil
 	}
 
+	// Single tenant deployment is tied to an explicit stack ID
```

Comment:
as an alternative, this could be split into it's own authorizer and not included in single tenant deploys

Reply:
ya -- not sure what is better, this should still check that the user org is 1 🤔

Reply:
> should still check that the user org is 1 
i'm not sure i follow. what i meant here was if `cfg.StackID != ""` then we inject the single tenant/org stack authorizer that runs the `if auth.stackID != "" {}` block

Reply:
yes -- i get that, but IIUC, in the case where `cfg.StackID != ""` we must *also* check that `signedInUser.OrgID == 1` 

See https://github.com/grafana/grafana/pull/76310/files#diff-6174ce995848073fe979f2143c2663a5fcbca42f1c200db47be5e244b6103ea5R55